### PR TITLE
Add manual module publish and tag workflow using FORGE_API_KEY secret

### PR DESCRIPTION
**Pull request by carabasdaniel**

### CODE REVIEW

**Comment by carabasdaniel** [comment]
Example run: https://github.com/carabasdaniel/puppetlabs-testing/actions/runs/617506533

File: moduleroot/.github/workflows/release.yml.erb [hunk 1]
```diff
@@ -0,0 +1,57 @@
+<% common = config_for('common') -%>
+name: "Publish module"
+
+on:
+  workflow_dispatch:
+    inputs:
+      forge_api_key:
+        description: "Forge API token to upload module to the Puppet Forge"
```
Comment (DavidS) [inline moduleroot/.github/workflows/release.yml.erb:8]: can we set this as github secret instead?

File: moduleroot/.github/workflows/release.yml.erb [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          ref: ${{ github.ref }}
+          clean: true
+      - name: "PDK Build"
+        uses: docker://puppet/pdk:nightly
+        with:
+          args: 'build'
+      - name: "Set token"
```
Comment (DavidS) [inline moduleroot/.github/workflows/release.yml.erb:51]:
```suggestion
      - name: "mask token"
```

**Review by DavidS — COMMENTED** [review]
one question inline

File: moduleroot/.github/workflows/release.yml.erb (hunk 1 above)
Comment (carabasdaniel) [reply moduleroot/.github/workflows/release.yml.erb:8]: Initial approach was to have it as a Github secret, but having it as an input might be easier for the future.
I'm open to suggestions here.

Comment (DavidS) [reply moduleroot/.github/workflows/release.yml.erb:8]: If you want to preserve it as input for testing/one-off releases, we could just do both and use `${FORGE_TOKEN:-${{ env.FORGE_TOKEN }}}` (see "Use Default Values" on https://linux.die.net/man/1/bash) at the call site

Comment (carabasdaniel) [reply moduleroot/.github/workflows/release.yml.erb:8]: Sounds like a good idea, I'll add that in

Comment (carabasdaniel) [reply moduleroot/.github/workflows/release.yml.erb]: Having issues masking the input token, so for initial workflow we'll use only the repository secret

**Review by adrianiurca — APPROVED** [review]

**Pull request merged**